src/test_rig_bluesky/plans.py

```diff
@@ -38,6 +51,44 @@ def spectroscopy(
     yield from bps.prepare(
         spectroscopy_detector, TriggerInfo(livetime=exposure_time), wait=True
     )
+    # TODO: This would be nicer if NDArrayBaseIO had a PortName signal
+    yield from bps.abs_set(spectroscopy_detector.fileio.nd_array_port, "D2.roistat")
+
+    rois = [
+        ROI(2, "Green", 880, 605, 150),
+        ROI(3, "Blue", 1665, 600, 150),
+        ROI(1, "Red", 95, 610, 150),
+    ]
+
+    params: list[NDAttributePv] = []
+    for roi in rois:
+        roistatn = spectroscopy_detector.roistat.channels[roi.channel]  # type: ignore
+        assert isinstance(roistatn, NDROIStatNIO)
+
+        yield from bps.abs_set(roistatn.name_, roi.name)
+        yield from bps.abs_set(roistatn.min_x, roi.start_x)
+        yield from bps.abs_set(roistatn.min_y, roi.start_y)
+        yield from bps.abs_set(roistatn.size_x, roi.size)
+        yield from bps.abs_set(roistatn.size_y, roi.size)
+        yield from bps.abs_set(roistatn.use, True)
+
+        # We can't include all the channels as it makes the xml longer than 256
+        # params.append(
+        params = [
+            NDAttributePv(
+                name=f"{roi.name}Total",
+                signal=roistatn.total,
+                dbrtype=NDAttributePvDbrType.DBR_LONG,
+                description=f"Sum of {roi.name} channel",
+            )
+        ]
+        # )
+
+    yield from setup_ndattributes(spectroscopy_detector.roistat, params)  # type: ignore
```

Comment:
Could: Extract this into a reusable `apply_rois` plan

Reply:
Actually I think we would be better off getting load/save working and then this code can
all be deleted.